src/components/query-browser.tsx

```diff
@@ -847,7 +847,7 @@ const QueryBrowser_: React.FC<QueryBrowserProps> = ({
             },
           );
           setGraphData(newGraphData);
-          onDataChange(newGraphData);
+          onDataChange?.(newGraphData);
```

Comment:
I'm curious why this didn't throw an TypeScript error? It's correctly marked as optional in `QueryBrowserProps`.